Kernel/Arch/aarch64/Interrupts.cpp

```diff
@@ -218,10 +225,44 @@ void initialize_interrupts()
 
 // Sets the reserved flag on `number_of_irqs` if it finds unused interrupt handler on
 // a contiguous range.
-ErrorOr<u8> reserve_interrupt_handlers([[maybe_unused]] u8 number_of_irqs)
+ErrorOr<u8> reserve_interrupt_handlers(u8 number_of_irqs)
 {
-    TODO();
-    return Error::from_errno(EINVAL);
+    bool found_range = false;
+    u8 first_irq = 0;
+    SpinlockLocker locker(s_interrupt_handler_lock);
+    for (size_t start_irq = 0; start_irq < s_interrupt_handlers.size(); start_irq++) {
+        auto*& handler_slot = s_interrupt_handlers[start_irq];
+        VERIFY(handler_slot != nullptr);
+
+        if (!is_unused_handler(handler_slot))
+            continue;
+
+        found_range = true;
+        for (auto off = 1; off < number_of_irqs; off++) {
+            auto*& handler = s_interrupt_handlers[start_irq + off];
+            VERIFY(handler_slot != nullptr);
+
+            if (!is_unused_handler(handler)) {
+                found_range = false;
+                break;
+            }
+        }
+
+        if (found_range == true) {
+            first_irq = start_irq;
+            break;
+        }
+    }
+
+    if (!found_range)
+        return Error::from_errno(EAGAIN);
+
+    for (auto irq = first_irq; irq < number_of_irqs; irq++) {
+        auto*& handler_slot = s_interrupt_handlers[irq];
+        handler_slot->set_reserved();
+    }
+
+    return first_irq;
```

Comment:
Maybe add a FIXME to share this code with x86-64, just like for other functions in this file.